src/flow/assembly_lmh.hh

```diff
@@ -104,12 +108,13 @@ public:
         if (genuchten_on) {
             conductivity=0;
             head=0;
+            std::vector<LongIdx> indices(ad_->dh_cr_->max_elem_dofs());
```

Comment:
Create the vector fo dofs in the DHCell and return the reference to it by get_loc_dof_indices, this simplifies the assembly and remove repeated allocation of the temporary vector.

Reply:
- introduce a class ConstArray<T>, constructed from C-pointer to T and size
- have size method, const begin and const end, operator[]
- change get*indices of to return ConstArray using directly indices stored in the dof handler